frontend/src/ManageGroupPermissions/ManageGroupPermissions.jsx

```diff
@@ -356,3 +561,41 @@ class ManageGroupPermissionsComponent extends React.Component {
 }
 
 export const ManageGroupPermissions = withTranslation()(ManageGroupPermissionsComponent);
+
+const Field = ({
```

Comment:
One thing as a suggestion. this file is growing bigger. it is better to move single components or modals into separate files and it will be more readable others.